### PR TITLE
Simplify functions

### PR DESCRIPTION
**Pull request by K-ballo**
Simplify `function` and `unique_function`. Unify duplicated functionality. Drop support for generic archive types.

### CODE REVIEW

**Comment by hkaiser** [comment]
Thanks! LGTM

File: hpx/util/function.hpp [hunk 1]
```diff
@@ -87,47 +78,4 @@
...
-            hpx::util::detail::function_vtable_ptr<
-                Sig
-              , hpx::serialization::input_archive, hpx::serialization::output_archive
-            >
-          , hpx::util::detail::empty_function<Sig>
-        >
-    >
-      : boost::mpl::false_
-    {};
-}}
```
Comment (sithhell) [inline hpx/util/function.hpp]: The purpose of this code was to allow successful compilation when automatic registration is switched of, which we sometimes need to ensure proper serialization between different platforms. Is this still working?

Comment (K-ballo) [inline hpx/util/function.hpp]: This moved into `basic_function.hpp`, to apply to all serializable functions. If it was working before then it should be working now.

Comment (sithhell) [inline hpx/util/function.hpp]: Gotcha, thanks!

**Pull request merged**